website/docs/docs/building-a-dbt-project/metrics.md

```diff
@@ -90,6 +92,85 @@ metrics:
+| name        | A unique identifier for the metric                          | new_customers                   | yes       |
+| model       | The dbt model that powers this metric                       | dim_customers                   |     yes (no for `expression` metrics)    |
+| label       | A short for name / label for the metric                     | New Customers                   | no        |
+| description | Long form, human-readable description for the metric        | The number of customers who.... | no        |
+| type        | The type of calculation to perform when evaluating a metric | count_distinct                  | yes       |
+| sql         | The expression to aggregate/calculate over                  | user_id                         | yes       |
+| timestamp   | The time-based component of the metric                      | signup_date                     | yes       |
+| time_grains | One or more "grains" at which the metric can be evaluated   | [day, week, month]              | yes       |
+| dimensions  | A list of dimensions to group or filter the metric by       | [plan, country]                 | no        |
+| filters     | A list of filters to apply before calculating the metric    | See below                       | no        |
+| meta        | Arbitrary key/value store                                   | {team: Finance}                 | no        |
+
+</VersionBlock>
+
+<VersionBlock firstVersion="1.2">
+
+### Available types
+
+| Metric Type    |  Description                                                               |
+|----------------|----------------------------------------------------------------------------|
+| count          | This metric type will apply the `count` aggregation to the specified field |
+| count_distinct | This metric type will apply the `count` aggregation to the specified field, with an additional distinct statement inside the aggregation |
+| sum            | This metric type will apply the `sum` aggregation to the specified field |
+| average        | This metric type will apply the `average` aggregation to the specified field |
+| min            | This metric type will apply the `min` aggregation to the specified field |
+| max            | This metric type will apply the `max` aggregation to the specified field |
+| expression     | This metric type is defined as any **non-aggregating** calculation of 1 or more metrics |
+
+</VersionBlock>
+
+<VersionBlock firstVersion="1.2">
+
+### Expression Metrics
+In v1.2, support was added for `expression` metrics, which are defined as non-aggregating calculations of 1 or more metrics. By defining these metrics, you are able to create metrics like:
+- ratios
+- subtractions 
+- any arbitrary calculation
+
+As long as the two+ base metrics (the metrics that comprise the `expression` metric) share the specified `time_grains` and `dimensions`, those attributes can be used in any downstream metrics macro.
```

Comment:
[Non-blocking]

> As long as the two+ base metrics (the metrics that comprise the `expression` metric) share the specified `time_grains` and `dimensions`

Ah - do we have validation for this in `dbt-core`? In the `dbt-metrics` package?

Reply:
We have "validation" for this in the `dbt_metrics` package. The macro constructs a list of acceptable dimensions from all metrics used and then the provided dimensions are compared against that list. The macro errors out if it is an invalid dimension.